File: src/data/scripts/weapons/aEP_ThrusterAnimation.java
package data.scripts.weapons;

import com.fs.starfarer.api.AnimationAPI;
import com.fs.starfarer.api.Global;
import com.fs.starfarer.api.combat.*;
import com.fs.starfarer.api.graphics.SpriteAPI;
import combat.util.aEP_Tool;
import org.lazywizard.lazylib.FastTrig;
import org.lazywizard.lazylib.MathUtils;
import org.lazywizard.lazylib.VectorUtils;
import org.lwjgl.util.vector.Vector2f;

import java.awt.*;

public class aEP_ThrusterAnimation extends MagicVectorThruster
{

  private static float maxFlameState = 30;// 0 == non flame, control flame length
  private ShipEngineControllerAPI.ShipEngineAPI e;
  private float amount = 0f;
  private float ORIGINAL_HEIGHT = 80f;
  private float ORIGINAL_WIDTH = 10f;

  public boolean enable = false;

  @Override
  public void advance(float amount, CombatEngineAPI engine, WeaponAPI weapon) {
    if(enable){
      super.advance(amount,engine,weapon);
    }
  }

}

        	         	       	       	 

File: src/data/scripts/weapons/aEP_ThrusterAnimation.kt
package data.scripts.weapons

import com.fs.starfarer.api.combat.ShipEngineControllerAPI.ShipEngineAPI
import com.fs.starfarer.api.combat.CombatEngineAPI
import com.fs.starfarer.api.combat.WeaponAPI
import com.fs.starfarer.api.util.Misc
import data.scripts.utils.aEP_ID
import data.scripts.utils.aEP_Tool
import org.magiclib.weapons.MagicVectorThruster
import java.awt.Color

class aEP_ThrusterAnimation : MagicVectorThruster() {
  private val e: ShipEngineAPI? = null
  private val amount = 0f
  private val ORIGINAL_HEIGHT = 80f
  private val ORIGINAL_WIDTH = 10f
  var enable = false
  override fun advance(amount: Float, engine: CombatEngineAPI, weapon: WeaponAPI) {
    if (enable) {
      super.advance(amount, engine, weapon)
      engine.addSmoothParticle(weapon.location,Misc.ZERO,40f,1f,aEP_Tool.getAmount(null),ENGINE_GLOW)
    }
  }

  companion object {
    private const val maxFlameState = 30f // 0 == non flame, control flame length
    val ENGINE_GLOW = Color(240,235,170,255)
  }
}